### ISSUE TITLE
RESOLVED - Using OwnTracks with openHABianPi on home LAN

### ISSUE THREAD

**Issue by meingraham** [owntracks/talk#45]
I have my openHAB 2.2.0 node set up on a Raspberry Pi 3B running openHABian connected to my home network. I have configured the openHAB Cloud Connector. I have mosquitto and the MQTT binding set up. All of this is functioning properly. I have also installed the OwnTracks binding.

My question is how to set up the OwnTracks mobile app so that it will send its location messages to my private MQTT broker. I understand that I need to "expose" my broker's IP and MQTT port so that OwnTracks can see it. I also understand that I need to secure this connection to protect against malicious access (this is the case anyway).

I can't believe I'm the first to want to set it up this way. But I have searched for several days and have not been able to find anything for my scenario. Everything seems to point to an MQTT broker on the Internet. Is there a configuration guide (detailed please since I'm a noob) available?

**Comment by jpmens** [comment]
This is clearly described in our Booklet.

**Comment by meingraham** [comment]
I looked through the booklet so I must be misunderstanding something. Could
you provide the link & bookmark to the pertinent section. I'm not trying to
shortcut doing the research. I just didn't see the documentation as
applying to my scenario.


Michael Ingraham
Spartanburg, SC

**Comment by meingraham** [comment]
I think the part I'm missing is the DNS & port forwarding setup. Those
"terms" are about as much as I know. The how to is what I'm not clear on...
and I've not found a good guide to hold my hand through it. That is, define
somewhere out on the Internet where I can point OwnTracks to a domain and
in turn that domain gets resolved by a DNS (DDNS probably, as if I know
that any better!). Then, the port forwarding on my router. Maybe I'm
getting confused by the term "forwarding". I can't get my head around how I
tell my router to allow traffic destined to my MQTT broker port to be
allowed through, 1. regardless of the source IP, or better, 2. the specific
source IP (preferably via a domain so as to cover the bases in case the IP
changes over time) for OwnTracks.

I'd be happy to get to that point and then deal with security. I tried
setting up TLS and generating the certificates and the keys. I tried this
numerous times testing w/in my own LAN and have not yet been able to
successfully communicate with my MQTT broker using TLS. So, for now I'm
using user/pwd security. Baby steps.


Michael Ingraham
Spartanburg, SC

**Comment by jpmens** [comment]
Admittedly we don't cover all those bits because they're not directly related to OwnTracks on the one hand, and because they'd fill seventeen volumes of book, on the other. :-)

You've got the terms right. Our apps will connect to your broker. In order to do that your router needs a fixed IP address or, as you've correctly determined, some dynamic DNS name which resolves to your changing IP address at home. I think the best people to ask might be your ISP.

In terms of "forwarding through your router", that's normally a setting in the router in which you tell it to let traffic for a particular port (1883 MQTT or 8883 MQTT over TLS) to the "inside", i.e. into your network. You then specify to which IP address/port combination for your broker.

**Comment by meingraham** [comment]
@jpmens I'm baaaack!

This is how long it's taken me working practically around the clock (not kidding) since we last corresponded four days ago. I now understand and set up a "stable" DDNS for my WAN IP, stood up a reverse proxy using NGINX with SSL certificates for my domain, and set up port forwarding. Phew! I'm using port 8883 on the Internet and the reverse proxy is delivering that to the mosquitto broker on my server. I have verified that port 8883 is open on my host (the reverse proxy). I still haven't tested that the proxy is doing its job properly of passing the traffic to mosquitto. But I wanted to generate some traffic. Thus I want OwnTracks to be able to successfully connect. Right now it isn't. The status message reads 'java.io.IOException: stream does not represent a PKCS12 key store'. And this leads to my question about SSL certificates and keys. 

I used Certbot to generate the certificates. It generated four certificate (PEM) files:

1. privkey.pem (obviously not this one)
2. fullchain.pem contains all certificates, including server certificate (aka leaf certificate or end-entity certificate)
3. cert.pem contains the server certificate by itself
4. chain.pem contains the additional intermediate certificate or certificates that web browsers will need in order to validate the server certificate.

This whole certificate nomenclature has me baffled. So, my next question... OwnTracks is asking for the CA certificate and the Client certificate. Which of these pem certificates applies to which OwnTracks certificate?

Thanks.

Mike

**Comment by jpmens** [comment]
Welcome baaack, @meingraham. I had quietly hoped I'd given you so much to learn that you'd be away a tad longer ... No worries: just kidding! :-)

Before I dive into your question, there are two words that scare me in your tale: "reverse proxy". Did you set up a reverse proxy as in an HTTP reverse proxy such as Apache with mod_proxy or nginx? If so, that will not work, unfortunately. MQTT (also MQTT over TLS) has nothing to do with HTTP(S) and as such will not flow over an HTTP proxy (reverse or otherwise), unless you're using Websockets (which I don't assume you are).

**Comment by jpmens** [comment]
To answer your question nevertheless: The CA certificate OwnTracks for Android is asking you, that's the content of fullchain.pem, respectively the first certificate in that file. (PEM files are editable and contain public keys as certificates between BEGIN/END lines (these lines belong to the PEM file). The client certificate OwnTracks wants you *don't* have, as certbot cannot create client certificates (yet) [it is my understanding that Let's Encrypt will at some state possibly do that, but it's not yet on the horizon)

Client certificates are for, e.g. authenticating a client to a server. Forget that for the time being, b/c you cannot do that with Let's Encrypt.

What I think you want is TLS (what used to be called SSL) encryption of data so that your smart phone transmits your username/password and all other OwnTracks data securely and in encrypted for between your device and the server. 

What you need for that is a Mosquitto broker set up and configured for TLS (I understand you're almost done there, if we can remove the "reverse proxy" doubt :) and you then need to configure your device with the server's CA (certification authority) certificate. That will suffice to accomplish what you need. I hope that's a wee bit clearer?

**Comment by meingraham** [comment]
@jpmens Thanks for the certificate clarification. I did have fullchain in the first one but had also put it in the second. I removed the second one. A different error (Endpoint State Error, 'Unexpected error (6)', Endpoint queue 0). I haven't had time to research. I'll look into the certificate config further later.

As for the reverse proxy - right. Quite a bit of stuff out there to reverse proxy MQTT protocol using NGINX. Some appear to require their commercial software. But I did find one setup (the one I have set up at this time) that proxies MQTT traffic via websockets (i.e., HTTPS).
[MQTT through nginx and owntracks](https://community.home-assistant.io/t/mqtt-through-nginx-and-owntracks/4943/13) which links to [MQTT Bridge with Mosquitto and nginx](http://frankfurtlovesyou.com/posts/mqtt-bridge-with-mosquitto-and-nginx.html).

I'll let you know if I ever get it working.

**Comment by jpmens** [comment]
I’d like to recommend you omit the proxy from your setup — it will cause pain and complexity, neither off which you really want or need.

What we generally do is configure a TLS-enabled Mosquitto broker on the “inside” (i.e. on the home network, let’s assume 192.168.1.1) and ensure that’s working including ACLs, etc.

Second step is then to “drill” a hole through your firewall. What this means is you look at your user manual and find the spot where you pass port 8883 from the Internet (TCP mind you) in to your internal network, address 192.168.1.1 port 8883, TCP. That will allow you (and everybody else who finds the router address) to access your Mosquitto broker. At this point you know the outside IP address (what’s my ip address?) of your router and could do a first test.

Next up, you find a dyndns service and set up a DNS name which points to your external router IP address; this can sometimes change, and your router will then update the dyndns service, the end result being you have a name “

**Comment by meingraham** [comment]
@jpmens - Yeah, I'm not married to a reverse proxy. I just want a solution that lets ONLY my traffic through to my MQTT broker.

I have my DDNS set up pointing to my public IP. I have a job on my server that keeps that refreshed in case my IP ever changes.

My server has a static IP on my LAN 192.168.1.nnn.

On my router I have port 8883 forwarded to my server. Right now I forward it to port 8883 on my reverse proxy but I can change this to point to 1883 on my server so MQTT traffic is forwarded to my broker.

You mentioned an Access Control List. Is this different than a firewall rule?

Would I have to set up a firewall on my server (which is what I was trying to accomplish with the reverse proxy) that only let's traffic from one known source IP address (OwnTracks). Is this doable? Would the OwnTracks IP address (or set of IP addresses) ever change and make my firewall rule quit working? Or worse, would one of those IP addresses ever get assigned to someone/something else?

How can I ensure that if someone crawls every IP and every port and finds my_Domain_or_IP:8883 that they can't attack my MQTT broker? I'm not well versed enough with this kind of configuration.

Regards.

Mike

**Comment by jpmens** [comment]
A firewall rule typically lets traffic from a set of addresses pass or a particular type of traffic. You’ve already guessed that that won’t be possible because your phone gets a *different* IP address depending on where you switch it on (WiFi, mobile carrier, diferent WiFi, etc.)

An Mosquitto ACL lets a particular client (say your OwnTracks client on the phone) either publiish or receibve or both on particular topics. Say, you and I share your broker, I publish to a topic owntracks/jpmens/phone and you may subscribe to that topic, thus your’re allowed to see where I am.

**Comment by jpmens** [comment]
The hex codes: you’re attempting to push MQTT protocol through an HTTP proxy. That won’t work (as I mentioned earlier).

**Comment by meingraham** [comment]
> The hex codes: you’re attempting to push MQTT protocol through an HTTP proxy. That won’t work (as I mentioned earlier).

So much for that [article I linked](https://community.home-assistant.io/t/mqtt-through-nginx-and-owntracks/4943/13) about routing MQTT to websockets... Maybe I just haven't set up Mosquitto to accept websockets. I'll have to read up on the config.

> A Mosquitto ACL lets **_a particular client_** (say your OwnTracks client on the phone) either publish or receive or both on particular topics.

So, if I understand, Mosquitto can be configured to only respond to certain client IDs. Otherwise, even if the topics are right, it ignores the content. Again, it looks like I need to dig deeper into the Mosquitto docs.

What I don't understand is, if someone wants to be malicious and pumps a ton of validly formatted MQTT traffic to myIP:8883, my broker may be kept busy "ignoring" all the bad traffic. Sure, nobody can see the OwnTracks location data we're exchanging, but someone could still perform something similar to a DDoS on my broker; yes?

**Comment by jpmens** [comment]
Yes

**Comment by meingraham** [comment]
@jpmens Progress! OwnTracks sending data with SSL in place making it through to my MQTT broker. Still lots of setup to be done (Regions, ACLs, openHAB config, etc.) But the 1st baby step and I'm still standing ;-)

Question: Certbot certificates expire after 90 days. A daily job is set up to check and renew. Does this mean that I have to send the new fullchain.pem to the OwnTracks client? I'm afraid that the answer is yes; right? If so, is there a way to create "non-expiring" certificates even if it's not using Certbot?

**Comment by jpmens** [comment]
Oh, no, definitely not. The chain you have installed on the device can validate the server certificate presented by your MQTT broker when the client connects, and that's all that's needed. So, to be quite clear, once the connection is working, which apparetnly now is the case, you won't have to do anything client-side when renewing your server-side certificate using Certbot.

**Comment by meingraham** [comment]
Sweet!

And it only took me 9 days to figure it out ;-)

**Comment by jpmens** [comment]
Eight and a half days on the Bahamas followed by an hour or two of research? Not bad! :-)

**Comment by jpmens** [comment]
@sumnerboy12 might I bother you to give Mike here (and myself of course) a hand with the openHAB mqttitude binding? I don't see the problem -- probably tomatoes on me eyes

**Comment by sumnerboy12** [comment]
I can't see any pubs arriving on owntracks/user/device/*event* in your logs - but I may be missing something? Have you got sufficient ACLs to allow those pubs? openHAB uses these /event pubs, not the location pubs.

**Comment by meingraham** [comment]
@sumnerboy12 Ben,

As @jpmens can attest, I've had a devil of a time getting SSL configured properly! So I'm going to tighten up security AFTER I get the setup working. That is to say, I have no ACLs configured currently.

To confirm that the data from OwnTracks is getting to the broker, I have run mosquitto_sub on the same node where mosquitto is running and I have also brought up a client using MQTTfc on another Windows PC connected to our LAN. I can see the data on both clients.

I set up mosquitto_sub to subscribe to -t '$SYS/#' and -t   '#'. In the logs above, most of the output is related to $SYS... but these are the lines for data published by OwnTracks:
```
03.05.2018 11:11:10   owntracks/<USER>/<owntracksDEVICE#1>   {"_type":"location","tid":"u1","acc":17,"batt":94,"conn":"w","lat":<LAT>,"lon":<LON>,"t":"u","tst":1525360269,"_cp":true}
03.05.2018 11:11:16   owntracks/<USER>/<owntracksDEVICE#1>   {"_type":"location","tid":"u1","acc":17,"batt":94,"conn":"w","lat":<LAT>,"lon":<LON>,"tst":1525360275,"_cp":true}
```

Since my earlier post, I've been playing with the case sensitivity of my configuration. I have confirmed that I have the <USER>, <owntracksDEVICE>, and each region description using the exact cased strings in both the Android app config as well as the openHAB items file. I did find that I was missing the <USER> in the items bindings (the openHAB MQTTITUDE binding documentation example just shows the device - I assume it's for a connection that does not use authentication). I still haven't got this working though. My items file now contains:
```
Switch   Presence<U1>_<REGION 1>   "<U1> <REGION 1>"   { mqttitude="openhab_tcpbroker:owntracks/<USER>/<owntracksDEVICE#1>/event:<REGION 1>" }
Switch   Presence<U1>_<REGION 2>   "<U1> <REGION 2>"   { mqttitude="openhab_tcpbroker:owntracks/<USER>/<owntracksDEVICE#1>/event:<REGION 2>" }
Switch   Presence<U1>_<REGION 3>   "<U1> <REGION 3>"   { mqttitude="openhab_tcpbroker:owntracks/<USER>/<owntracksDEVICE#1>/event:<REGION 3>" }
Switch   Presence<U2>_<REGION 1>   "<U2> <REGION 1>"   { mqttitude="openhab_tcpbroker:owntracks/<USER>/<owntracksDEVICE#2>/event:<REGION 1>" }
Switch   Presence<U2>_<REGION 2>   "<U2> <REGION 2>"   { mqttitude="openhab_tcpbroker:owntracks/<USER>/<owntracksDEVICE#2>/event:<REGION 2>" }
Switch   Presence<U2>_<REGION 3>   "<U2> <REGION 3>"   { mqttitude="openhab_tcpbroker:owntracks/<USER>/<owntracksDEVICE#2>/event:<REGION 3>" }
```
I can't find it again, but somewhere in the OwnTracks booklet I read that I ought to be using `use_identity_as_username`. With all the security configuration issues I'd been having, I hadn't used option. I've set a "static" username because I wasn't sure how I'd set up the openHAB MQTT binding nor the MQTTITUDE binding if I used that option. I wouldn't think this is the problem, but I'll defer to your experience with this.

Cheers!

Mike

**Comment by meingraham** [comment]
Just saw a few events for my wife's phone:
```
{"_type":"transition","tid":"u2","acc":142.5,"event":"leave","lat":<LAT>,"lon":<LON>,"tst":1525385348,"wtst":1525300290,"t":"c"}
{"_type":"transition","tid":"u2","acc":37.741,"event":"leave","lat":<LAT>,"lon":<LON>,"tst":1525385591,"wtst":1525300331,"t":"c"}
{"_type":"transition","tid":"u2","acc":39.003,"event":"enter","lat":<LAT>,"lon":<LON>,"tst":1525397025,"wtst":1525300331,"t":"c"}
{"_type":"transition","tid":"u2","acc":39.003,"event":"enter","lat":<LAT>,"lon":<LON>,"tst":1525397025,"wtst":1525300290,"t":"c"}
```
I still don't see that my switches changed state. And since I haven't moved, openHAB is oblivious that I'm inside my Home geofence... even though my phone is sending OwnTracks location data.

**Comment by jpmens** [comment]
When we say _event_ we mean a transition event which happens when a device enters or leaves a region (the docs sometimes erroneously term those waypoints). E.g. your home, your office, etc. That’s what we call an event.

Location data (`_type: location`) is transmitted when the app thinks it wants to. Depending on iOS/Android there are different modes. Typically this happens when the operating system informs the apps that you’ve moved a significant distance which can be around 500m. Ah, you say “Android”, so yes, approximately that.

The apps do not periodically send transition events — it’’s exclusively on crossing a border into / out of your region.

The place to ask questions about openHAB is on the forum. However, I will disclose a little-known secret: our very own @sumnerboy12 is he who authored the _mqttitude_ binding for openHAB! Do check the community there though; maybe your issue has been addressed already.

**Comment by sumnerboy12** [comment]
Your `mqttitude` binding config looks fine. And if your apps are publishing `/event` messages then you should be seeing those openHAB switches changing state. Is there logs being generated by the `mqttitude` binding? Probably worth enabling TRACE logging for that binding and seeing what is going on.

**Comment by meingraham** [comment]
I received a few transition events from my phone. I got these messages in the log immediately following each transition message from my phone:
```
06-May-2018 12:13:45.196 [TRACE] [enhab.binding.mqttitude.internal.MqttitudeConsumer] - Checking item SPT_house_VT_<uTag1>_region_town...
06-May-2018 12:13:45.198 [TRACE] [enhab.binding.mqttitude.internal.MqttitudeConsumer] - Location enter event has no region (**missing or empty 'desc'**), ignoring
```

I had shared the regions on my wife's phone but not on my phone. So, now finally, I think I have all the settings configured properly.

I'll have to check this the next time we go away from the house.

I didn't get any from my wife's phone, yet we were together... and I did receive some location messages from her phone. I don't understand why I didn't get any transition messages from my wife's phone though. I haven't changed her settings and I was getting them a couple of days ago.

**Comment by meingraham** [comment]
@sumnerboy12 @jpmens 

Something is not right... obviously!

I've been running the TRACE log in openHAB for a couple of days now. It has received a few transition events from my phone when I've left the house. It has yet to receive a transition message from my wife's phone. I looked at the Android notification log history on her phone, and there are notifications from OwnTracks showing that it sent transition events (I can look at the body of the notification and see the text "Leaving <region>". In addition, I have an MQTT client running on another computer on my network and it has received transition event topics from my wife's phone. openHAB is receiving all OwnTracks messages from my phone as well location messages from my wife's phone.

I tried to find a way to archive the Android notification log - no success. I managed to save some screenshots.

[owntracks.log](https://github.com/owntracks/talk/files/1982018/owntracks.log)
![6522682214040396396](https://user-images.githubusercontent.com/34340210/39733842-9f5ba022-5242-11e8-9e6f-1e505190cd9c.png)
![5289939137484352831](https://user-images.githubusercontent.com/34340210/39733843-9f6bcdb2-5242-11e8-9a31-b561d7c6ccad.png)
![7653627525008079886](https://user-images.githubusercontent.com/34340210/39733844-9f77c52c-5242-11e8-8115-fe67dc3ba535.png)
[mqtt_event_topics.txt](https://github.com/owntracks/talk/files/1982020/mqtt_event_topics.txt)

**Comment by sumnerboy12** [comment]
Are you seeing your openHAB switches get _switched_ when *your* phone sends an event transition? I.e. is it working for your phone at least?

**Comment by sumnerboy12** [comment]
It is very hard to parse the pages and pages of logs and text you have posted, but it looks like the Owntracks app is not sending the region name in the payload, which I believe happens if the region is not marked public/shared. Without that name/desc openHAB can't determine which item to switch, as indicated by the trace logs.

**Comment by meingraham** [comment]
Ben,

> @sumnerboy12 "It is very hard to parse the pages and pages of logs"

I debated stripping out extraneous bits of the log... but then I decide that perhaps there was something I would not see in the logs that might provide you with a clue.

The Regions were shared on both devices, I swear! I just checked my phone and indeed they aren't shared. OK, that explains the lack of description for my events.

However, as can be seen in the payloads for my wife's phone, those messages contain the description. Nevertheless, those messages don't even seem to have been seen by openHAB... yet the location messages do trigger the manual configuration.

**Comment by meingraham** [comment]
@sumnerboy12 Ben,

Back to basics... I'm obviously missing something core that is just eluding me.

I uninstalled the binding, followed steps to ensure that indeed no traces of it were left by looking directly through the openHAB Console.

I reinstalled the binding that came with the openHABian image.

I captured some transactions again and merged the mqtt messages and the MQTTITUDE trace output. 
[mqttitude tracing](https://docs.google.com/spreadsheets/d/1t8AwBcuneMGyUXe7uce0JsWq23PxPqMVgA7hzy5GAO0)

I matched the MQTT messages to the MQTTITUDE log messages. You'll see that the location messages are triggering the appropriate MQTTITUDE actions but nothing is happening for region transition messages.


What is strange is that, aside from the fact that my phone did not have its region definitions shared, the MQTTITUDE log does show that it was "responding" to transition messages. Of course it was not able to successfully complete the action because there was no description in the message. Nevertheless, at least we know that MQTTITUDE is attempting to process the message. Yet, for "fully formed" transition messages from my wife's phone, the log does not show it taking any action at all.

I saw that there were some newer SNAPSHOTs of the MQTTITUDE binding here:

- Snapshot build: [https://openhab.ci.cloudbees.com/job/openHAB-Distribution](https://openhab.ci.cloudbees.com/job/openHAB-Distribution)
- Pull Request build: [https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/](https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/)

Should I be using one from one of these repositories?

Thanks

Mike

**Comment by sumnerboy12** [comment]
The MQTTITUDE binding hasn't changed in a long time (unless someone else has been pushing changes without my knowledge, which is entirely possible!). It is a relatively simple binding so there is not much to change tho. If you have *region* items configured (i.e. `/event`) then the binding will process those messages and ignore if there is no waypoiint/region name in the payload.

There is obviously something amiss with your config and my guess is it is on the phone app side. Since you seem to have your MQTT broker happily receiving messages, and the MQTTITUDE binding is processing those messages. 

Sorry I can't be of more help but there is nothing obviously wrong from what I can see.

**Comment by meingraham** [comment]
@sumnerboy12 @jpmens

I'm still working to resolve my issue. I started with a clean slate and set up only MQTT & MQTTITUDE. Still have the issue.

Still trying to think of what I may be doing wrong... some questions to see if there is conflict between my OwnTracks definitions:

- I have a "manual" definition set up in my mqttitude.cfg. Because I was following the binding documentation, I've ended up calling that definition "home" (all lowercase).
- Are region names case sensitive? I'm assuming yes. I have the name (i.e., the description) of my regions propercased (first letter capitalized). I have three concentric regions: 'Home', 'Town', and 'Away'. I use the same syntax in my openHAB Items definition.
- Is there any potential conflict between the 'Home' region and the 'home' manual definition? They are both using the same lat/lon and radius.

Regarding region "behavior". R1 within R2 which in turn is within R3.
- Are the regions concentric "bands" or overlapping circles? If I 'leave' R1 do I 'enter' R2 **and** R3 or just R2?
- How does OwnTracks differentiate a transition when leaving R3 and entering R2 vs. leaving R3 as I go further away?

**Comment by jpmens** [comment]
I’d like to suggest you start simple. None of this R1, R2, R3 stuff. Just a single region into which / out of which you move to see if your openHAB gets it. I don’t know regarding the case sensitivity: in case (hah!) of doubt, keep them identical.

Track your phone with `mosquitto_sub` to your broker, and make sure you’re seeing the correct location and event publishes when entering or leaving a region. BTW, don’t make the region too small: a radius of 100m or more should trigger quite nicely.

Once that works, add complexity if need be.

**Comment by meingraham** [comment]
Thanks JP.

Simplification sounds logical. I'll delete the "manual" entry in mqttitude.cfg and define only one region on each device for now. It'll be tomorrow at the earliest to see if this results in any change to the behavior I'm seeing (transition messages from my wife's phone not being "seen").

Regarding initialization. How will openHAB know where each device is until each one crosses the region boundary? Am I right to suspect that MQTT persistence will send the last transition message from each device when Mosquitto starts up?

My top priority is pretty simple. I want to define two groups: Anybody Home (**any** of the devices inside the radius), Everyone Away (**all** of the devices outside the radius).

**Comment by meingraham** [comment]
@jpmens 

One region on each device. Transition messages being published are arriving (confirmed via mosquitto_sub). No action logged by mqttitude for messages from my wife's device.

**Comment by meingraham** [comment]
@jpmens,

There was suggestion that I change the Device ID on my wife's phone to see if that had any effect. Taking that one step further, I decided to uninstall the app and then reinstall it. Blowing the app away and setting everything back up resolved the issue. I should of done this as part of my "start from scratch" approach. I only took the start over step on my openHAB server. I'm kind of glad I didn't because at least now I know where the problem came from... a bad string in one of the entries on my wife's phone (probably the Device ID since the user was used to authenticate and that was working).

Perhaps there was an extraneous non-printable character in one of those fields that I could not "see" in the MQTT topic that was keeping the openHAB binding from resolving what appeared to be the exact same name.

Frustrating to troubleshoot for sure! In any case, it appears to be finally resolved.

Thanks for following up!

Mike

**Comment by jpmens** [comment]
Glad you solved it

**Comment by meingraham** [comment]
@jpmens 

I'm afraid I've done something wrong. We had this exchange back on May 2 regarding the certificate on the mobile device. I'd asked whether I had to "renew" the certificate on the mobile device when the certificates where renewed on the server.

> The chain you have installed on the device can validate the server certificate presented by your MQTT broker when the client connects, and that's all that's needed. So, to be quite clear, once the connection is working, which apparently now is the case, you won't have to do anything client-side when renewing your server-side certificate using Certbot.

I just noticed that OwnTrack app updates were not updating in my openHAB server. Well, the certificates renewed this past week... and the status in the OwnTracks app on my mobile device shows that the connection has failed.

Which certificate do I need to import on the mobile device? Is it one of the certificates Certbot generated for my domain or is it the trusted CA from LetsEncrypt? If the latter, where do I obtain that certificate? In any case, which certificate do I need to import on the mobile device so that I don't have to renew it every time I renew the server certificates?

Thanks!

Mike

**Comment by jpmens** [comment]
The certificate you need on the mobile device is the _trusted CA from LetsEncrypt_ and you get that via their Web site. However, unless you've removed certificates from your device you shouldn't need to install it. 

If you installed the site certificate (generated by Certbot) just remove that; you should be able to, on the device, see that its date has expired.

**Comment by meingraham** [comment]
@jpmens 

There are two "certificate" field entries in the OwnTracks app: CA certificate and Client certificate. I have left the Client certificate blank but did specify my own CA certificate (using my Certbot fullchain.pem). I do need the CA certificate; right? And I can just use the LetsEncrypt CA certificate for that; right?

I just wanted to be sure I understood specifically.

Thanks.

Mike

**Comment by jpmens** [comment]
You do not have your own CA certificate as you're using Let's Encrypt. (*They* are the CA and their CA certificate is trusted already by your device, or should be.) What you have is a server certificate and you do not need to install that on your device.

**Comment by meingraham** [comment]
@jpmens 

Thanks JP.

What I meant by "my own" was one which Certbot generate on my server based on my domain. True, I am not a Certificate Authority... I'm neither an authority nor am I trusted ;-)

Cheers!

Mike
